src/libexpr/primops.cc

```diff
@@ -2271,18 +2271,22 @@ static void prim_removeAttrs(EvalState & state, const Pos & pos, Value * * args,
     state.forceList(*args[1], pos);
 
     /* Get the attribute names to be removed. */
-    std::set<Symbol> names;
-    for (auto elem : args[1]->listItems()) {
+    Symbol names[args[1]->listSize()];
+    for (auto [idx, elem] : enumerate(args[1]->listItems())) {
         state.forceStringNoCtx(*elem, pos);
-        names.insert(state.symbols.create(elem->string.s));
+        names[idx] = state.symbols.create(elem->string.s);
     }
 
+    auto it = &names[0], end = &names[args[1]->listSize()];
+    std::sort(it, end);
```

Comment:
You’re not really `O(r+n)` (like f17684cca5c6c8afa469af9ad38026410de24da2 says) I assume since you have to sort. More `O(r*log(r)+n)`. Still faster than the existing implementation though.

Reply:
yeah, the `r` there refers to the sorted set, not the unsorted input list. :sweat_smile: